From 003c2e84d95357bbbef72fb395ef85f1650097a3 Mon Sep 17 00:00:00 2001
From: Adrian Perez de Castro <aperez@igalia.com>
Date: Fri, 23 Sep 2022 14:35:51 +0300
Subject: [PATCH] Meson: Un-hardcode building a shared library

Use library() instead of shared_library() to allow specifying which kind
of library to build with "meson setup --default-library ..."

This allows more easily incorporating libmanette in an application e.g.
as a Meson subproject which gets built into a program as a static
library.

Signed-off-by: Adrian Perez de Castro <aperez@igalia.com>
[Upstream status: https://gitlab.gnome.org/GNOME/libmanette/-/merge_requests/84]
---
 src/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/meson.build b/src/meson.build
index c140cac..b71d090 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -63,7 +63,7 @@ if gudev.found()
   libmanette_deps += [ gudev ]
 endif
 
-libmanette_lib = shared_library(
+libmanette_lib = library(
   libmanette_module,
   libmanette_sources,
   c_args: libmanette_c_args,
-- 
2.37.3

